### PR TITLE
Rules Style guide

### PR DESCRIPTION
**Pull request by WilcoFiers**

### CODE REVIEW

File: pages/style-guide.md [hunk 1]
```diff
@@ -0,0 +1,79 @@
+---
+title: Auto-WCAG Rules Style Guide
+---
+
+This style guide provides instructions on how to formulate and format your rules. **[Learn how to design a rule](rule-design.html)**, before starting with this guide.
```
Comment (rdeltour) [inline pages/style-guide.md]: the `rule-design` page doesn't exist yet, but shouldn't this link point the the Markdown version so that it's navigable in the GitHub UI too?

File: pages/style-guide.md [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+
+- All block level components (lists, paragraphs, headings, tables, etc.) should be separated by a **single** empty line. Kramdown (unlike GitHub) occasionally concatenates tables and lists to other items if this isn't done.
+
+- References to HTML elements and attributes should always use the back ticks character, followed by either the word element(s) or attribute(s). Example: **\`div\` element** and **\`alt\` attribute`**
+
+## Procedural Statements
+
+Each rule has a number of steps, where each step is made up of several statements. For example: 
+
+	If the selected image has no alt, continue with [Step 2](#step-2).
```
Comment (rdeltour) [inline pages/style-guide.md]: "no `alt` attribute", to be consistent with the formatting style above ? 😉

File: pages/style-guide.md [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+	If true, return:
+
+
+### Variables
+
+TODO
+
+## Steps In A Test Procedure
+
+The test procedure is broken up into steps toA make the rule easier to read and understand. Each step is often a single if-statement, possibly including a few statements that have to be executed before the if statement. As a rule of thumb, hold that each step in a rule should do one of the following three things:
```
Comment (rdeltour) [inline pages/style-guide.md]: s/toA/to/

File: pages/style-guide.md [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+
+In order to maximize readability, don't use direct links in paragraphs. Use `[link text][ref]` instead of `[link text](url)`. *(Note the brackets are different.)* When a link stands on it's own, or is the only text in a list item, this is not a problem. Links used multiple times on the same page should also use references.
+
+The reference should be a abbreviation of no more than 6 characters, written in all capitals. The expanded reference is placed together with the URL at the bottom of the document, like so:
+
+	[TXTALT]: ../pages/algorithms/text-alt.html
+
+## Providing Notes
+
+TODO
```
Comment (rdeltour) [inline pages/style-guide.md]: Also add a  section on "helper" functions? (e.g. "is empty or whitespace", etc)

File: pages/style-guide.md [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+
+- References to HTML elements and attributes should always use the back ticks character, followed by either the word element(s) or attribute(s). Example: **\`div\` element** and **\`alt\` attribute`**
+
+## Procedural Statements
+
+Each rule has a number of steps, where each step is made up of several statements. For example: 
+
+	If the selected image has no alt, continue with [Step 2](#step-2).
+
+These statements are often no more then two are three sentences long. They are written in plain English
```
Comment (EmmaJP) [inline pages/style-guide.md]: two "or" three steps (word correction)

File: pages/style-guide.md [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+	
+	If the selected element has an `alt` attribute, continue with [Step 3](#step-3)
+
+	Else return:
+
+	| Outcome  | Failed
+	|----------|-----
+	...
+
+The term *selected element* should be used to refer to the each item that is obtained by the selector.
```
Comment (EmmaJP) [inline pages/style-guide.md]: refer to each item (the not needed)

File: pages/style-guide.md [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+
+When rules get complicated, it becomes useful to give the step a short name that is descriptive of what the step does. This makes it easier to understand the flow of a rule. Example: 
+
+	### Step 1: check element type (F65)
+
+## Referencing Other Documents
+
+In order to maximize readability, don't use direct links in paragraphs. Use `[link text][ref]` instead of `[link text](url)`. *(Note the brackets are different.)* When a link stands on it's own, or is the only text in a list item, this is not a problem. Links used multiple times on the same page should also use references.
+
+The reference should be a abbreviation of no more than 6 characters, written in all capitals. The expanded reference is placed together with the URL at the bottom of the document, like so:
```
Comment (EmmaJP) [inline pages/style-guide.md]: ...an abbreviation ... written all in capitals. (minor corrections)
